w3_d4_2_js_integration_testing/README.md

```diff
@@ -0,0 +1,85 @@
+# API Testing with Javascript
+
+| Objectives |
+| :--- |
+| Add basic test coverage to any app through integration tests |
+| Use Test Driven Development and Behavior Driven Development |
+
+| Concepts | Tools | Activities |
+| :---: | :---: | :---: |
+| Integration Testing TDD/BDD | Mocha, Supertest, Chai | Challenges |
+
+### Preparatory Readings
+
+1. [Difference Between Test Methodologies](http://stackoverflow.com/questions/4904096/whats-the-difference-between-unit-functional-acceptance-and-integration-test/4904533#4904533)
+
+
+### Motivation (Why?)
+
+### Analogy (What?)
+
+### Key Concepts & Snippets
+
+#### Unit & Integration Tests
+
+**Unit Tests** test only one part of the code independently.
+
+**Integration Tests** test all the code the way a user would use it.
+
+#### Example of a Simple API Test
+
+```
+var request = require('request')
+  , expect = require('chai').expect
+
+var baseUrl = 'http://localhost:3000';
+
+describe('home page should say "hello world!"', function() {
+  request(baseUrl, function (error, response, body) {
+    expect(body.to.contain('hello world!'));
+  });
+});
+```
+
+#### Example of Chai Expect Selectors
+
+```
+```
+
+#Challenges
+
+### Docs & Resources
+
+* **[Mocha](http://mochajs.org/#getting-started)** - for running asynchronous tests
+* **[Request](https://github.com/request)** - for handling HTTP request/response
+* **[Chai](http://chaijs.com/api/)** - for assertions
+
+### Basic Challenges
+
```

Comment:
Can you be more specific here? Have you previously explained what a GET test is?